File: IOT/Anshul-13J_AnshulNigam_2024cs1069_2/FunctionAndArrays/AnyBaseSubtraction.cpp
#include<iostream>
using namespace std;

int getDifference(int b, int n1, int n2){
    int borrow=0,diff=0,d1[3]={},d2[3]={},i=0,j=0,l=0,d[3],k;
    while(n2)
    {
        d2[i++]=n2%10;
        n2/=10;
    }
    while(n1)
    {
        d1[j++]= n1%10;
        n1/=10;
    }

    for(k=0 ; k<3 ;k++)
    {
        if(d2[k]<d1[k])
        {
            d2[k]+=b;
            d2[k+1]-=1;
        }
        d[l++]=d2[k]-d1[k];
    }
    for(i=2;i>=0;i--)
    {
        diff= diff*10 + d[i];
    }
    cout<<diff;
    return 0;
}


int main(){
    int b, n1, n2;
    cin>>b>>n1>>n2;
    getDifference(b,n1,n2);
}

File: IOT/anshulgurjar53_AnshulSinghGurjar_2024it1193_2/FunctionAndArrays/anybasesubtraction.java
import java.util.*;
  
  public class Main{
  
  public static void main(String[] args) {
      Scanner scn = new Scanner(System.in);
      int b = scn.nextInt();
      int n1 = scn.nextInt();
      int n2 = scn.nextInt();
  
      int d = getDifference(b, n1, n2);
      System.out.println(d);
   }
  
   public static int getDifference(int b, int n1, int n2){
       int rem1=0,rem2=0,x=0,c=0,quo=0,borrow=0,num=0;
       while(n1!=0||n2!=0||borrow!=0)
       {
           rem1=n1%10;
           rem2=n2%10;
           rem2+=borrow;
           //x=rem1+rem2+carry;
           if(rem2<rem1)
           {
               borrow=-1;
               rem2+=b;
               x=rem2-rem1;
               quo=x*(int)Math.pow(10,c++);
           }
           else{
               borrow=0;
               x=rem2-rem1;
               quo=x*(int)Math.pow(10,c++);
           }
           
           
           num+=quo;
           n1=n1/10;
           n2=n2/10;
       }
       return num;

   }
  
}